PiwikTracker/URLSessionDispatcher.swift

```diff
@@ -12,46 +12,44 @@ final class URLSessionDispatcher: Dispatcher {
     let session: URLSession
     let baseURL: URL
 
-    var userAgent: String? = {
-        #if os(OSX)
-            let webView = WebView(frame: .zero)
-            let currentUserAgent = webView.stringByEvaluatingJavaScript(from: "navigator.userAgent") ?? ""
-        #elseif os(iOS)
-            let webView = UIWebView(frame: .zero)
-            let currentUserAgent = webView.stringByEvaluatingJavaScript(from: "navigator.userAgent") ?? ""
-        #elseif os(tvOS)
-            let currentUserAgent = ""
-        #endif
-        return currentUserAgent.appending(" PiwikTracker SDK URLSessionDispatcher")
-    }()
+    private(set) var userAgent: String?
     
     /// Generate a URLSessionDispatcher instance
     ///
     /// - Parameters:
     ///   - baseURL: The url of the piwik server. This url has to end in `piwik.php`.
     ///   - userAgent: An optional parameter for custom user agent.
-    init(baseURL: URL, userAgent: String? = nil) {
-        if !baseURL.absoluteString.hasSuffix("piwik.php") {
-            fatalError("The baseURL is expected to end in piwik.php")
-        }
+    init(baseURL: URL, userAgent: String? = nil) {                
         self.baseURL = baseURL
         self.timeout = 5
         self.session = URLSession.shared
-        if (userAgent != nil) {
-            self.userAgent = userAgent
+        asyncMain {
+            self.userAgent = userAgent ?? URLSessionDispatcher.defaultUserAgent()
```

Comment:
This this is not 100% thread safe since we might access self.userAgent on something else then the main queue.

Reply:
I thinks that's fine so far. There's barely a change that this is false. Even if it will be corrected with the next request. In the end, the User Agent is "just" needed to extract some information about the device, such as the OS. If the first request lacks the data, the next one will add it, which will fill in the blank first spots.